### PR TITLE
refactor(launcher): use class syntax

### PR DESCRIPTION
**Pull request by devoto13**

### CODE REVIEW

File: lib/launcher.js [hunk 1]
```diff
@@ -26,13 +26,36 @@ const baseBrowserDecoratorFactory = function (
...
-  let lastStartTime
+class Launcher {
+  constructor (server, emitter, injector) {
+    this._server = server
+    this._emitter = emitter
+    this._injector = injector
+    this._browsers = []
+    this._lastStartTime = null
+
+    this.launch.$inject = [
```
Comment (johnjbarton) [inline lib/launcher.js]: This bit is real mysterious to most readers. Can we add a comment like:
```
// Attach list of dependency injection parameters to methods.
```

File: lib/server.js [hunk 2]
```diff
@@ -75,7 +75,7 @@ class Server extends KarmaEventEmitter {
       emitter: ['value', this],
       server: ['value', this],
       watcher: ['value', watcher],
-      launcher: ['type', Launcher],
+      launcher: ['factory', Launcher.factory],
```
Comment (johnjbarton) [inline lib/server.js]: Could this be an API change, requiring new major version rather then refactor?

File: lib/launcher.js (hunk 1 above)
Comment (devoto13) [reply lib/launcher.js]: Done

File: lib/server.js (hunk 2 above)
Comment (devoto13) [reply lib/server.js]: No, it should not. This only changes how new instance is created and consumers shouldn't normally create new launcher themselves. How launcher is injected stays exactly the same and should not require any changes in plugins.

**Comment by devoto13** [comment]
Travis passed, but status on GitHub was not updated for some reason.

**Pull request merged**